src/test/clone/test_fork.rs

```diff
@@ -1697,13 +1690,121 @@ fn main() -> Result<(), Box<dyn Error>> {
+                // asm block, effectivey hiding the double-return.
+                unsafe {
+                    core::arch::asm!(
+                        // vfork
+                        "syscall",
+                        // If in the parent, exit the asm block (by jumping forward
+                        // to the label `2`).
+                        // https://doc.rust-lang.org/rust-by-example/unsafe/asm.html#labels
+                        "cmp rax, 0",
+                        "jne 2f",
+
+                        // We're in the child; execve syscall.
+                        "mov rax, r12",
+                        "syscall",
+                        "ud2",
+
+                        "2:",
+                        inlateout("rax") libc::SYS_vfork => raw_pid,
+                        // r12 shouldn't be clobbered by the vfork syscall
+                        in("r12") libc::SYS_execve,
+                        in("rdi") path.as_ptr(),
+                        in("rsi") execv_argvec(&args).as_ptr(),
+                        in("rdx") execv_argvec(&env).as_ptr(),
+                        clobber_abi("C"),
+                    )
+                };
+                Pid::from_raw(raw_pid.try_into().unwrap()).unwrap()
+            }),
+        ),
+        (
+            "posix_spawn",
+            Arc::new(|path: &Path, args: &[&str]| {
+                let mut raw_pid = 0;
+                let path = CString::new(path.as_os_str().as_bytes()).unwrap();
+                let args: Vec<CString> = [path.clone()]
+                    .into_iter()
+                    .chain(args.iter().map(|s| CString::new(*s).unwrap()))
+                    .collect();
+                let env: Vec<CString> = Vec::new();
+                let rv = unsafe {
+                    libc::posix_spawn(
+                        &mut raw_pid,
+                        path.as_ptr(),
+                        std::ptr::null(),
+                        std::ptr::null(),
+                        execv_argvec(&args).as_ptr().cast(),
+                        execv_argvec(&env).as_ptr().cast(),
+                    )
+                };
+                assert_eq!(rv, 0);
+                Pid::from_raw(raw_pid).unwrap()
+            }),
+        ),
+        (
+            "rust_command",
+            Arc::new(|path: &Path, args: &[&str]| {
+                let args: Vec<OsString> = args
+                    .iter()
+                    .map(|s| OsString::from_str(s).unwrap())
+                    .collect();
+                let child = std::process::Command::new(path)
+                    .args(&args)
+                    .spawn()
+                    .unwrap();
+                let pid = Pid::from_raw(child.id().try_into().unwrap()).unwrap();
+                // Our API doesn't support returning the Child object;
+                // forget it so that its Drop doesn't try to kill or wait for it.
+                std::mem::forget(child);
```

Comment:
The docs say:

> There is no implementation of Drop for child processes, so if you do not ensure the Child has exited then it will continue to run, even after the Child handle to the child process has gone out of scope.

So I don't think this is needed?

Reply:
done